### PR TITLE
added functions for local storage

### PR DESCRIPTION
**Pull request by nhieubrian**
# Pull Request Process
1. Ensure that your branch is as up-to-date with main as possible 
2. Fill out this template
3. Have at least 2 other people to get PR approval comments (could just add everybody on team)
4. Create PR

# Changes
## What changes did you make?
- Added some more CSS code for resizing issues
- Added JS functions to call local storage information
- Temporary function in place to parse data for a set recipe
- **Added a function from Lab 6, need confirmation if we can use that**

# Verification
## How did you ensure that your changes work?

- Used a recipe example to test the local storage functions
- Resized locally
- Birthday cake image is grabbed from local storage, rather than saved or original json.

# Screenshots
## Please post screenshots of your changes working if possible

559d8475c45252beed8b5d31710ddba7

### CODE REVIEW

File: source/user-portal/assets/favRecipe.js [hunk 1]
```diff
@@ -76,24 +77,44 @@ class FavRecipe extends HTMLElement {
...
+        else{
+
+            let json = JSON.parse(localStorage.getItem(data));
+
+            // Adding Title
+            const text = document.createTextNode(data);
+            recipeTitle.appendChild(text);
+
+            // Adding Img
+            let value = searchForKey(json, 'thumbnailUrl');
```
Comment (nlpham) [inline source/user-portal/assets/favRecipe.js:110]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8436327)
```suggestion
            let value = searchForKey(json, "thumbnailUrl");
```

File: source/user-portal/assets/script.js [hunk 2]
```diff
@@ -1,9 +1,7 @@
 // Script for adding recipe card
 
-// This would be the array from Local Storage
 const recipes = [
-
-
+    'https://introweb.tech/assets/json/birthdayCake.json'
```
Comment (nlpham) [inline source/user-portal/assets/script.js:4]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8436327)
```suggestion
    "https://introweb.tech/assets/json/birthdayCake.json"
```

File: source/user-portal/assets/favRecipe.js [hunk 3]
```diff
@@ -104,5 +125,28 @@ class FavRecipe extends HTMLElement {
...
+ * @returns {*} the value of the found key
+ */
+ function searchForKey(object, key) {
+    var value;
+    Object.keys(object).some(function (k) {
+      if (k === key) {
+        value = object[k];
+        return true;
+      }
+      if (object[k] && typeof object[k] === 'object') {
```
Comment (nlpham) [inline source/user-portal/assets/favRecipe.js:144]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8436327)
```suggestion
      if (object[k] && typeof object[k] === "object") {
```

File: source/user-portal/assets/script.js [hunk 4]
```diff
@@ -12,45 +10,61 @@ window.addEventListener('DOMContentLoaded', init);
...
 
         for(let i = 0; i < recipes.length; i++){
             fetch(recipes[i])
-            .then((response) => response.json())
-            .then((data) => {
-                recipeData[i] = data;
-                if(i + 1 == recipes.length){
-                    resolve(true);
-                }
+                .then(response => response.json())
```
Comment (nlpham) [inline source/user-portal/assets/script.js:33]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8436327)
```suggestion
            .then((response) => response.json())
```

File: source/user-portal/assets/favRecipe.js [hunk 5]
```diff
@@ -76,24 +77,44 @@ class FavRecipe extends HTMLElement {
...
+            // Adding Title
+            const text = document.createTextNode(data);
+            recipeTitle.appendChild(text);
+
+            // Adding Img
+            let value = searchForKey(json, 'thumbnailUrl');
+            recipeImg.setAttribute("src", value);
+
+            // Adding Time
+            value = searchForKey(json, 'totalTime');
```
Comment (nlpham) [inline source/user-portal/assets/favRecipe.js:114]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8436327)
```suggestion
            value = searchForKey(json, "totalTime");
```

File: source/user-portal/assets/script.js [hunk 6]
```diff
@@ -12,45 +10,61 @@ window.addEventListener('DOMContentLoaded', init);
...
+                .then(data =>{
```
Comment (nlpham) [inline source/user-portal/assets/script.js:34]: Codacy has a fix for the issue: [Missing space after =>.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8436327)
```suggestion
                .then(data => {           
```

Comment (nlpham) [inline source/user-portal/assets/script.js:34]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Big-Bytez/cse110-fa21-group19/pullRequest?prid=8436327)
```suggestion
            .then((data) => {
```

**Review by limpa105 — APPROVED** [review]
This looks great to me! My only question is what do you mean by PTO? I have been just using minutes for the recipies do you think we should use something else?

**Comment by nhieubrian** [comment]
It is just how the recipe JSON formatted their total time. Minutes will be fine.

**Review by nlpham — APPROVED** [review]
Cards look way better.

**Pull request merged**